src/app/child-dev-project/reporting/reporting/reporting.component.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, OnInit } from "@angular/core";
+import { ActivatedRoute } from "@angular/router";
+import { Aggregation, ReportingService, ReportRow } from "../reporting.service";
+
+export interface ReportingComponentConfig {
+  aggregationDefinitions?: Aggregation[];
+}
+
+@Component({
+  selector: "app-reporting",
+  templateUrl: "./reporting.component.html",
+  styleUrls: ["./reporting.component.scss"],
+})
+export class ReportingComponent implements OnInit {
+  config: ReportingComponentConfig;
+  results: ReportRow[];
+  fromDate: Date;
+  toDate: Date;
+  reportTable: { label: string; result: any }[];
+
+  constructor(
+    private activatedRoute: ActivatedRoute,
+    private reportingService: ReportingService
+  ) {}
+
+  ngOnInit() {
+    this.activatedRoute.data.subscribe((config) => (this.config = config));
+  }
+
+  async calculateResults() {
+    this.reportingService.setAggregations(this.config.aggregationDefinitions);
+    this.results = await this.reportingService.calculateReport(
+      this.fromDate,
+      this.toDate
+    );
+
+    // The ReportRowComponents manipulates the values array of the elements
+    // Using a timeout the calculation waits for this manipulation to be done
+    setTimeout(() => (this.reportTable = this.getExportTable()));
```

Comment:
where does it change data? I can't see it (at least when [values] input is not used)